ecs-agent/acs/session/task_manifest_responder.go

```diff
@@ -0,0 +1,177 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"sync"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type TaskManifestResponder interface {
+	New(taskComparer TaskComparer, messageIDAccessor ManifestMessageIDAccessor,
+		snAccessor SequenceNumberAccessor) wsclient.RequestResponder
+}
+
+type TaskComparer interface {
+	CompareRunningTasksOnInstanceWithManifest(*taskManifestResponder, *ecsacs.TaskManifestMessage) ([]*ecsacs.TaskIdentifier, error)
+}
+
+const (
+	TaskManifestMessageName = "TaskManifestMessage"
+)
+
+// taskManifestResponder responds to task manifest messages from ACS. It processes the
+// message and, if the message isn't stale, creates a list of stop candidates, which are
+// tasks that are running on the instance but are tasks that ACS says should not be running.
+// The taskManifestResponder will ack the task manifest message and send ACS a
+// TaskStopVerificationMessage if there are stop candidates.
+type taskManifestResponder struct {
+	taskComparer      TaskComparer
+	snAccessor        SequenceNumberAccessor
+	messageIDAccessor ManifestMessageIDAccessor
+	cluster           string
+	respond           wsclient.RespondFunc
+	logger            *logrus.Entry
+	metricsFactory    metrics.EntryFactory
+}
+
+// NewTaskManifestResponder returns an instance of the taskManifestResponder struct.
+func NewTaskManifestResponder(taskComparer TaskComparer, snAccessor SequenceNumberAccessor, messageIDAccessor ManifestMessageIDAccessor,
+	cluster string, metricsFactory metrics.EntryFactory, responseSender wsclient.RespondFunc) *taskManifestResponder {
+	r := &taskManifestResponder{
+		taskComparer:      taskComparer,
+		snAccessor:        snAccessor,
+		messageIDAccessor: messageIDAccessor,
+		cluster:           cluster,
+		logger:            logrus.WithField("ACSRequestResponder", "TaskManifestResponder"),
```

Comment:
Other responder types seem to be using logger provided by `github.com/aws/amazon-ecs-agent/ecs-agent/logger` package. Any reason we are using `logrus` here?

Reply:
A quick vscode search lead me to think the agent used logrus but it seems to be indirect, updating with logger soon 👍